kong/core/router.lua

```diff
@@ -128,18 +151,41 @@ local function marshall_api(api)
     if #api.uris > 0 then
       api_t.match_rules = bor(api_t.match_rules, MATCH_RULES.URI)
 
-      for i, uri in ipairs(api.uris) do
-        local escaped_uri = [[\Q]] .. uri .. [[\E]]
-        local strip_regex = escaped_uri .. [[\/?(.*)]]
-
-        api_t.uris[uri] = {
-          strip_regex = strip_regex,
-        }
+      for _, uri in ipairs(api.uris) do
+        if re_match(uri, [[^[a-zA-Z0-9\.\-_~/%]*$]]) then
+          -- plain URI or URI prefix
+          local escaped_uri = [[\Q]] .. uri .. [[\E]]
+          local strip_regex = escaped_uri .. [[/?(?P<stripped_uri>.*)]]
```

Comment:
the `P` isnt necessary; i wonder if removing it will help keep things cleaner:

```
$ cat t.lua 
local m, err = ngx.re.match("hello, world", "[a-z](?<name>[a-z]+)", "ajo")
print(m[0])
print(m[1])
print(m.name)
$ resty ./t.lua 
hello
ello
ello
```